kv/memberlist/memberlist_client.go

```diff
@@ -1184,7 +1184,10 @@ func (m *KV) mergeValueForKey(key string, incomingValue Mergeable, casVersion ui
 	m.storeMu.Lock()
 	defer m.storeMu.Unlock()
 
-	curr := m.store[key].Clone()
+	// Note that we do not take a copy of the store value here, it is modified in-place.
```

Comment:
I got confused here, since `curr` is a copy.
Would it be clearer to write "we do not take a copy of `curr.value`"?
Or to say "deep copy" ?